docs/style-spec/expressions-examples.md

```diff
@@ -0,0 +1,298 @@
+## Highway shields
+
+Icon assignment based on shield design and reference number length.
+
+Example output value: 
+`us-interstate-3`
+
+```
+{
+	"icon-image": {
+		"expression" : [
+				"concat",["string", ["get", ["properties"], "shield"]], "-", ["string", ["get", ["properties"], "ref"]]
+		]
+	}
+}
+```
+
+## Mountain peak labels
+
+Mountain peak label displays name and elevation if elevation value exists, otherwise just name. 
+
+```
+{
+	"text-field": {
+		"expression": [
+			"concat",
+			["string", ["get", ["properties"], "name_en"]],
+			["case",
+				["has", ["properties"], "elevation_m"],
+				["concat", "\n ", ["string",["get", ["properties"], "elevation_m"]]],
+				""
+			]
+		]
+	}
+}
+```
+
+## Landuse polygon colors
+
+```
+{
+	"fill-color": {
+		"expression": [
+			"curve", "linear", ["zoom"],
+				15.5, 
+				["match", ["get", ["properties"], "class"], 
+					"park", "hsl(100, 58%, 76%)",
+					"school", "hsl(50, 47%, 81%)",
+					"hospital", "hsl(340, 37%, 87%)"
+				],
+				16,
+				["match", ["get", ["properties"], "class"], 
+					"park", "hsl(100, 58%, 76%)",
+					"school", "hsl(50, 63%, 84%)",
+					"hospital", "hsl(340, 63%, 89%)"
+				],
+		]
+	}
+}
+```
+
+## City labels: case study across multiple properties
+
+At low zoom levels, city labels are often displayed on basemaps with text and a locator dot. The labels are often different sizes (depending on size of city), with text offset in different directions for optimal placement, and national capitals indicated by a dot with a border:
+
+![screen shot 2017-06-14 at 7 35 15 pm](https://user-images.githubusercontent.com/5607844/27159403-bd37c7a6-513a-11e7-9bfb-f3fa0047fe5e.png)
+
+At higher zoom levels, city labels are usually *just* text: 
+
+![screen shot 2017-06-14 at 7 37 08 pm](https://user-images.githubusercontent.com/5607844/27159404-bd3b39fe-513a-11e7-811d-7b15f830a97b.png)
+
+To create the style above using Mapbox Streets Source vector tiles requires:
+
+Expressions for the following style properties:
+
+- `icon-image`
+- `text-size`
+- `text-justify`
+- `text-anchor`
+- `text-offset`
+
+Based on the following data properties:
+
+- `scalerank`: size/prominence of city
+- `capital`: capital city status, where `capital:2` is a national capital
+- `ldir`: pre-determined optimum label direction
+
+```
+{
+	"icon-image": {
+		"expression": [
+			"case",
+			["<=", ["number", ["get", ["properties"], "scalerank"]], 2],
+			["case", 
+				["==", ["number", ["get", ["properties"], "capital"]], 2],
+				"capital-city-dot-large",
+				"city-dot-large" 
+			],
+			["<=", ["number", ["get", ["properties"], "scalerank"]], 5],
+			["case", 
+				["==", ["number", ["get", ["properties"], "capital"]], 2],
+				"capital-city-dot-medium",
+				"city-dot-medium" 
+			],	
+			[">", ["number", ["get", ["properties"], "scalerank"]], 5],
+			["case", 
+				["==", ["number", ["get", ["properties"], "capital"]], 2],
+				"capital-city-dot-small",
+				"city-dot-small" 
+			]
+		]	
+	}
```

Comment:
Ah, except for one thing: as of now, `step` curves (like their predecessor, `interval` functions) use intervals that are inclusive of the left-hand boundary and exclusive of the right-hand one.  We could make this configurable (`["step", ">"]` / `["step", ">="]`), or we could just rely on a workaround like using `2.001` instead of `2` as the step value.